core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServices.java

```diff
@@ -135,10 +136,37 @@
     }
 
     /**
+     * Returns {@link MediaType} specified by {@link ConsumeType} annotation.
+     */
+    private static MediaType consumeType(Method method) {
+        ConsumeType consumeType = method.getAnnotation(ConsumeType.class);
```

Comment:
Good suggestion. I think it's possible and would be useful. I will work on both of `ConsumeType` and `ProduceType`.